dotnet/src/Connectors/Connectors.AI.OpenAI/CustomClient/OpenAIClientBase.cs

```diff
@@ -114,111 +114,140 @@ private protected virtual void AddRequestHeaders(HttpRequestMessage request)
     /// </summary>
     private readonly HttpClient _httpClient;
 
-    private async Task<T> ExecutePostRequestAsync<T>(string url, string requestBody, CancellationToken cancellationToken = default)
+    private protected async Task<T> ExecutePostRequestAsync<T>(string url, string requestBody, CancellationToken cancellationToken = default)
     {
         HttpResponseMessage? response = null;
+        HttpContent? content = null;
```

Comment:
Neither of the declared variables is used outside the try block/scope. So, please declare and dispose them inside the block using this syntax: using var content = .....